### PR TITLE
chore: updated dependencies

### PR DESCRIPTION
**Pull request by webdeveric**
- Updated dependencies
  - `devDependencies` now require Node 14+
  - Updated GH Actions `node-version` due to `devDependencies` requirements
    - Removed `12.x`
    - Added `18.x`
    - The `tsconfig.json` still targets Node 12.
- Updated tests due to changes in dependencies
- Updated `eslint` config to allow `any` in test files

### CODE REVIEW

File: src/pack.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export async function pack(this: EsbuildServerlessPlugin) {
     const artifactPath = path.join(this.workDirPath, SERVERLESS_FOLDER, zipName);
 
     // remove prefixes from individual extra files
-    const filesPathList = pipe<IFiles, IFiles, IFiles>(
```
Comment (webdeveric) [inline src/pack.ts:123]: Not needed. TS will infer the type

**Comment by webdeveric** [comment]
e2e tests are passing for me locally but failing in GH Actions. Any idea why?

**Comment by samchungy** [comment]
> e2e tests are passing for me locally but failing in GH Actions. Any idea why?

https://github.com/floydspace/serverless-esbuild/blob/f60b9f59778f586747d61818612042991cfd43d7/.github/workflows/release.yml#L21
It might be to so with different versioning of packages. We should probably do `npm ci` instead of this

**Comment by webdeveric** [comment]
@samchungy Good idea, but that didn't fix it.

I updated the `pull.yml` action to upload an artifact on failure of `e2e` tests to aid in debugging.
I downloaded the artifact and compared the contents with what is built on my local Ubuntu machine.
The `.js` file contents are identical (looking under `e2e/individually/`).
The only diffs come from the `.json` files. The diffs are with the `S3Key` (expected) and `CodeSha256`.

**Comment by samchungy** [comment]
> @samchungy Good idea, but that didn't fix it.
> 
> I updated the `pull.yml` action to upload an artifact on failure of `e2e` tests to aid in debugging. I downloaded the artifact and compared the contents with what is built on my local Ubuntu machine. The `.js` file contents are identical (looking under `e2e/individually/`). The only diffs come from the `.json` files. The diffs are with the `S3Key` (expected) and `CodeSha256`.

Interesting, I'll have a look if I have time near the end of the week

**Comment by floydspace** [comment]
Hi @webdeveric I quickly checked your PR, and see you changed packager from `yarn` to `npm` in individually example for e2e testing, however this example was intended to show how to use `serverless-esbuild` with yarn also.
Today I made a commit fixing e2e tests for individually example, simply using yarn install for it in the `Makefile`

I see now this branch has conflicts, could you please resolve them, or I can do that if you cannot to do it shortly.
Cheers.

**Comment by webdeveric** [comment]
@floydspace I'll try to take a look at the conflicts tonight.

**Comment by webdeveric** [comment]
@floydspace I've fixed the conflicts and the build is passing now.

**Review by samchungy — APPROVED** [review]

**Review by floydspace — APPROVED** [review]
For some reason locally I get a different logical Id generated, not sure why and what is the logic behind serverless, I will not count on this, because the pipeline is green.
image
I added minor commit reverting not necessary .gitignore changes
Approve. Gonna squash and merge. Thank you for contribution

**Pull request merged**